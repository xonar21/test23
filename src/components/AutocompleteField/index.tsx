import React, { useState, useEffect } from "react";
import { FormControl, TextField, CircularProgress, Autocomplete, styled } from "@mui/material";
import { useDispatch } from "react-redux";
import { SelectListAction, SelectListSelector, SelectListsActions } from "~/store";

interface AutocompleteFieldProps {
  field: any;
  fieldValue: any;
  fieldKey: string;
  setValue: (newValue: any) => void;
  value: any;
  dinamicTitle: string;
  t: any;
  openModal: boolean;
  selectList: any;
  paramsRequest: any;
}

const AutocompleteField: React.FC<AutocompleteFieldProps> = ({
  field,
  fieldValue,
  fieldKey,
  setValue,
  value,
  dinamicTitle,
  t,
  openModal,
  selectList,
  paramsRequest,
}) => {
  const dispatch = useDispatch();
  const [page, setPage] = useState(1);
  const [loading, setLoading] = useState(false);
  const [totalPages, setTotalPages] = useState(0);
  const [searchValue, setSearchValue] = useState("");
  const actionKey: SelectListAction = field.action;
  const action = SelectListsActions[actionKey];
  const selectkey: SelectListSelector = field.keySelect;
  const [items, setItems] = useState(selectList[selectkey].items);
  const StyledListbox = styled("ul")({
    "&::-webkit-scrollbar": {
      width: "4px",
    },
    "&::-webkit-scrollbar-thumb": {
      backgroundColor: "rgba(0, 48, 92, 1)",
      borderRadius: "4px",
    },
    "&::-webkit-scrollbar-track": {
      backgroundColor: "rgba(230, 230, 230, 1)",
    },
  });

  useEffect(() => {
    const fetchData = async () => {
      if (field.action && !field.parent) {
        await dispatch(
          action(
            paramsRequest(page, 20, {
              fieldName: "value",
              values: [searchValue],
            }),
          ),
        );
      }
    };

    fetchData();
  }, [page, searchValue]);

  useEffect(() => {
    setTotalPages(selectList[selectkey].totalPages);

    if (selectList[selectkey].totalPages === 1) {
      setItems(selectList[selectkey].items);

      return;
    }

    setItems((prevItems: any) => [...prevItems, ...selectList[selectkey].items]);
  }, [selectList[selectkey].items]);

  useEffect(() => {
    const fetchData = async () => {
      if (field.parent && value[field.parent]) {
        const paramsRequest = (
          id: string,
          number?: number,
          size?: number,
          filters?: object,
          sortField?: string,
          sortOrder?: string,
        ) => {
          return {
            id,
            number,
            size,
            filters,
            sortField,
            sortOrder,
          };
        };

        await dispatch(
          action(
            paramsRequest(value[field.parent], page, 20, {
              fieldName: "value",
              values: [searchValue],
            }),
          ),
        );
      }
    };

    fetchData();
  }, [value[field.parent]]);

  useEffect(() => {
    if (field.parent) {
      setItems([]);
    }
  }, [openModal]);

  return (
    <FormControl fullWidth>
      <Autocomplete
        id="demo-autocomplete"
        options={items}
        value={fieldValue}
        getOptionLabel={option => {
          if (dinamicTitle === t("edit")) {
            let itemValue = "";
            items.forEach((e: any) => {
              if (e.key === option) {
                itemValue = e.value;
              }
            });
            return itemValue;
          }
          return option ? option.value : "";
        }}
        onChange={(e, newValue) => {
          setValue({ ...value, [fieldKey]: newValue?.key });
        }}
        onInputChange={(event, value, reason) => {
          if (reason === "reset") {
            setPage(1);
          }
          if (reason === "input") {
            setSearchValue(value);
            setPage(1);
            setItems([]);
          }
          if (reason === "clear") {
            setPage(1);
            setSearchValue("");
          }
          if (value === "") {
            setPage(1);
            setSearchValue("");
          }
        }}
        ListboxComponent={StyledListbox}
        ListboxProps={{
          onScroll: event => {
            const target = event.target as HTMLUListElement;
            const { scrollTop, scrollHeight, clientHeight } = target;
            if (scrollHeight - scrollTop === clientHeight) {
              setPage(prevPage => {
                if (prevPage !== totalPages) {
                  return prevPage + 1;
                }
                return prevPage;
              });
            }
          },
        }}
        renderInput={params => (
          <TextField
            {...params}
            label={t(`${field.fieldName}`)}
            InputLabelProps={{ ...params.InputLabelProps }}
            InputProps={{
              ...params.InputProps,
              endAdornment: (
                <>
                  {loading ? <CircularProgress size={20} /> : null}
                  {params.InputProps.endAdornment}
                </>
              ),
            }}
          />
        )}
        renderOption={(props, option) => <li {...props}>{option.value}</li>}
      />
    </FormControl>
  );
};

export default AutocompleteField;
