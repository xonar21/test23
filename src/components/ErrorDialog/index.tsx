import React from "react";
import Dialog from "@mui/material/Dialog";
import DialogTitle from "@mui/material/DialogTitle";
import DialogContent from "@mui/material/DialogContent";
import DialogContentText from "@mui/material/DialogContentText";
import DialogActions from "@mui/material/DialogActions";
import Button from "@mui/material/Button";

interface ErrorDialogProps {
  open: boolean;
  onClose: () => void;
  errorText: string;
}

const ErrorDialog: React.FC<ErrorDialogProps> = ({ open, onClose, errorText }) => {
  return (
    <Dialog open={open} onClose={onClose}>
      <DialogTitle>{"Ошибка"}</DialogTitle>
      <DialogContent>
        <DialogContentText>{errorText}</DialogContentText>
      </DialogContent>
      <DialogActions>
        <Button onClick={onClose} color="primary">
          Закрыть
        </Button>
      </DialogActions>
    </Dialog>
  );
};

export default ErrorDialog;
