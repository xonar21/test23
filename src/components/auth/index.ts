export * from "./auth-button";
export * from "./auth-form";
export * from "./auth-provider";
