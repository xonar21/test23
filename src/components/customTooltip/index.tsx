import React, { useMemo } from "react";
import { ColDef, ITooltipParams } from "ag-grid-community";
import { Box } from "@mui/material";

type CustomTooltipParams = ITooltipParams & {
  color: string;
  colDef: ColDef;
  shouldShowTooltip: boolean;
  rowData: any;
};
export default (props: CustomTooltipParams) => {
  const data = useMemo(() => props.api.getDisplayedRowAtIndex(props.rowIndex!)!.data, []);

  if (props && props.column) {
    if ("colId" in props.column) {
      const cellSelector = `.ag-row[row-id="${props.rowIndex}"] .ag-cell[col-id="${props.column.getColId()}"]`;
      const cellElement = document.querySelector(cellSelector);
      if (cellElement && cellElement) {
        if (cellElement.scrollWidth === cellElement.clientWidth) {
          return null;
        }
      }
    }
  }

  const subProperties = (data: any, props: any) => {
    const [firstField, secondField] = props.colDef.field.split(".");

    if (secondField) {
      return data[firstField][secondField];
    }

    return data[firstField];
  };
  return (
    <>
      {props.colDef.field ? (
        <Box
          sx={{
            backgroundColor: props.color || "white",
            border: "1px solid rgba(0, 48, 92, 1)",
            minWidth: "100px",
            borderRadius: "3px",
            padding: "5px",
            transition: "all 1s linear",
            maxWidth: "300px",
            whiteSpace: "pre-wrap !important",
            overflowWrap: "break-word !important",
            boxSizing: "border-box !important",
          }}
        >
          {subProperties(data, props)}
          {/* {data[props.colDef.field]} */}
        </Box>
      ) : (
        <></>
      )}
    </>
  );
};
