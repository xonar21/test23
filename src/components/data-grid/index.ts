export { default as DataGrid } from "./DataGrid";
export * from "./configs";
