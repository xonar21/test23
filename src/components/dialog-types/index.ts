export { default as DialogTypes } from "./dialog-types";
export * from "./dialog-types";
