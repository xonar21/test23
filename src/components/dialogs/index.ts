export * from "./prompt-dialog";
export * from "./post-form-dialog";
export * from "./dialog-provider";
