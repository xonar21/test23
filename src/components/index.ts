export * from "./auth";
export * from "./breadcrumbs";
export * from "./data-grid";
export * from "./demo";
export * from "./dialogs";
export * from "./dialog-types";
export * from "./formik";
export * from "./header";
export * from "./mui";
export * from "./navigation-drawer";
export * from "./page-loading-indicator";
export * from "./seo";
export * from "./snackbar-provider";
