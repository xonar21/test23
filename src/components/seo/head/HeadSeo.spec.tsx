import { HeadSeo } from "~/components";
import { render } from "~/shared";

describe("HeadSeo component", () => {
  it("should render without crashing", () => {
    const { asFragment } = render(<HeadSeo />);
    expect(asFragment()).toMatchSnapshot();
  });
});
