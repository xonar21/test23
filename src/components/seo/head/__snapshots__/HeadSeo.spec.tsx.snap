// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeadSeo component should render without crashing 1`] = `<DocumentFragment />`;
